ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamSinkBuilder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import io.confluent.ksql.execution.context.QueryContext;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.StreamSink;
+import io.confluent.ksql.execution.util.SinkSchemaUtil;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.serde.KeySerde;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+
+public final class StreamSinkBuilder {
+  private StreamSinkBuilder() {
+  }
+
+  public static void build(
+      final KStream<Object, GenericRow> kstream,
```

Comment:
wrong generics: `kstream` would be either `KStream<Windowed<Struct>, GenericRow>` or `KStream<Struct, GenericRow>`. Neither of which can be cast to `KStream<Object, GenericRow>`.

Maybe this method should be something like:

```java
  public static <K> void build(
      final KStream<K, GenericRow> kstream,
      final StreamSink<KStream<K, GenericRow>> streamSink,
      final KsqlQueryBuilder queryBuilder) {
```

???